src/components/Shodan/Shodan-API-Tool.tsx

```diff
@@ -1,51 +1,95 @@
-import { Button, LoadingOverlay, Stack, TextInput } from "@mantine/core";
+import { Button, Stack, TextInput } from "@mantine/core";
 import { useForm } from "@mantine/form";
-import { useCallback, useState } from "react";
+import { useCallback, useState, useEffect } from "react";
 import { CommandHelper } from "../../utils/CommandHelper";
 import ConsoleWrapper from "../ConsoleWrapper/ConsoleWrapper";
-import { UserGuide } from "../UserGuide/UserGuide";
+import { RenderComponent } from "../UserGuide/UserGuide";
 import { SaveOutputToTextFile_v2 } from "../SaveOutputToFile/SaveOutputToTextFile";
-
-const title = "Shodan API Tool";
-const description_userguide =
-    "The Shodan API is a powerful tool that allows external network scans to be performed with " +
-    "use of a valid API key. This key is obtained through account creation within Shodan; see the " +
-    "below link to create an account:\n\nShodan Account Creation: " +
-    "https://developer.shodan.io/api/requirements\n\nHow to use Shodan API:\n\n" +
-    "Step 1: Enter a Valid API Key - Note; See above for account creation to " +
-    "receive API Key.\n       E.g. PLACEHOLDER\n\nStep 2: Enter a Host IP.\n       " +
-    "E.g. 127.0.0.1\n\n" +
-    "Step 3: Click Scan to commence the Shodan API operation.\n\n" +
-    "Step 4: View the Output block below to view the results of the tool's execution.";
-
-interface FormValues {
+import { LoadingOverlayAndCancelButton } from "../OverlayAndCancelButton/OverlayAndCancelButton";
+import { checkAllCommandsAvailability } from "../../utils/CommandAvailability";
+import InstallationModal from "../InstallationModal/InstallationModal";
+
+/**
+ * Represents the form values for the Shodan API Tool component.
+ */
+interface FormValuesType {
     hostIP: string;
     shodanKey: string;
 }
 
+/**
+ * The Shodan API Tool component.
+ * @returns The Shodan component.
+ */
 export function ShodanAPITool() {
-    const [loading, setLoading] = useState(false);
-    const [output, setOutput] = useState("");
-    const [pid, setPid] = useState("");
-    const [allowSave, setAllowSave] = useState(false);
-    const [hasSaved, setHasSaved] = useState(false);
-
+    const [loading, setLoading] = useState(false); // State variable to indicate loading state.
+    const [output, setOutput] = useState(""); // State variable to store the output of the command execution.
+    const [pid, setPid] = useState(""); // State variable to store the process ID of the command execution.
+    const [allowSave, setAllowSave] = useState(false); // State variable to allow saving the output to a file.
+    const [hasSaved, setHasSaved] = useState(false); // State variable to indicate if the output has been saved.
+    const [loadingModal, setLoadingModal] = useState(false); // State variable to indicate loading state of the modal.
```

Comment:
You may need to change this to true. I am finding that the installation modal briefly pops up each time I go into the tool. Please investigate.

Reply:
> You may need to change this to true. I am finding that the installation modal briefly pops up each time I go into the tool. Please investigate.

Your 100% right. Strange behavior wasn't there last time I ran DDT with updated code. loadingModal needs to be set to true and the popup occurs only first time.. Just compiled and run code with set to true and now it only runs first time. Thank you for your very good help.